src/pages/[SbtAddress].tsx

```diff
@@ -0,0 +1,282 @@
+import {
+  useAccount,
+  useContract,
+  useStarknetCall,
+  useStarknetExecute,
+  useTransaction,
+} from "@starknet-react/core";
+import { NextPage } from "next";
+import { useRouter } from "next/router";
+import React, { ReactElement, useEffect, useState } from "react";
+import BN from "bn.js";
+import { ec, GetTransactionResponse, hash } from "starknet";
+import Button from "@/components/UI/button";
+import SelectIdentity from "@/components/selectIdentity";
+import Connect from "@/components/connection/connect";
+import styles from "@/styles/generator.module.css";
+import sbt_abi from "@/abi/starknet/sbt_abi.json";
+import { TextField } from "@mui/material";
+import LoadingScreen from "@/components/UI/screens/loadingScreen";
+
+type MetadataProps = {
+  name: string;
+  description: string;
+  image: string;
+};
+
+type CallDataProps = {
+  calldata: (string | number)[];
+  contractAddress: string;
+  entrypoint: string;
+};
+
+type TransactionDatas = {
+  status: string;
+};
+
+const SbtGenerator: NextPage = () => {
+  const router = useRouter();
+  const { SbtAddress } = router.query;
+
+  const { account, address } = useAccount();
+  const [contractAddress, setContractAddress] = useState<string>("");
+  const [privateKey, setPrivateKey] = useState<BN>();
+  const [tokenId, setTokenId] = useState<string>("0");
+  const [password, setPassword] = useState<string>("");
+  const [sbtData, setSbtData] = useState<MetadataProps>();
+  const [callData, setCallData] = useState<CallDataProps[]>();
+  const [element, setElement] = useState<ReactElement | null>(null);
+  const [loadingMessage, setLoadingMessage] = useState<string>("");
+  const [clickedMint, setClickedMint] = useState<boolean>(false);
+  const [transactionHash, setTransactionHash] = useState<string>("");
+  const [finalStep, setFinalStep] = useState<boolean>(false);
+  const {
+    data: transactionData,
+    loading: transactionLoading,
+    error: transactionError,
+  } = useTransaction({ hash: transactionHash });
+
+  const { execute } = useStarknetExecute({
+    calls: callData,
+  });
+  const { contract } = useContract({
+    address: contractAddress,
+    abi: sbt_abi,
+  });
+  const { data, loading, error } = useStarknetCall({
+    contract,
+    method: "get_uri",
+    args: [1],
+    options: {
+      watch: false,
+    },
+  });
+
+  useEffect(() => {
+    setContractAddress(SbtAddress as string);
+  }, [SbtAddress]);
+
+  useEffect(() => {
+    if (!loadingMessage) setElement(null);
+    setElement(<LoadingScreen message={loadingMessage} />);
+  }, [loadingMessage]);
+
+  useEffect(() => {
+    if (!loading) setElement(null);
+    if (address && loading && !error) setLoadingMessage("Fetching SBT data");
+  }, [loading, address]);
```

Comment:
I'll test to see how it returns in the front but I feel like you don't need these 2 `useEffect`, only one with a skeleton for example.

Reply:
I can't test cause it does not build as I told you on Telegram